web/src/test/java/org/bookkeepervisualmanager/cache/MetadataCacheTest.java

```diff
@@ -49,12 +50,16 @@ public void test() {
                 List<Long> ledgersInBookie = metadataCache.getLedgersForBookie("localhost:1234");
                 assertEquals(1, ledgersInBookie.size());
 
-                List<Ledger> ledgersByMeta = metadataCache.searchLedgers("foo", null);
+                List<Ledger> ledgersByMeta = metadataCache.searchLedgers("foo", null, null);
                 assertEquals(1, ledgersByMeta.size());
-                List<Ledger> ledgersByBookie = metadataCache.searchLedgers(null, "localhost:1234");
+                List<Ledger> ledgersByBookie = metadataCache.searchLedgers(null, "localhost:1234", null);
                 assertEquals(1, ledgersByBookie.size());
-                List<Ledger> ledgersByBookieAndMeta = metadataCache.searchLedgers("pulsar", "localhost:1234");
+                List<Ledger> ledgersByBookieAndMeta = metadataCache.searchLedgers("pulsar", "localhost:1234", null);
                 assertEquals(1, ledgersByBookieAndMeta.size());
+                List<Ledger> ledgersByBookieAndMetaAndIdOk = metadataCache.searchLedgers("pulsar", "localhost:1234", Arrays.asList(new Long[] {1L}));
+                assertEquals(1, ledgersByBookieAndMetaAndIdOk.size());
+                List<Ledger> ledgersByBookieAndMetaAndIdKo = metadataCache.searchLedgers("pulsar", "localhost:1234", new ArrayList());
```

Comment:
Collections.emptyList ?
please also some negative tests, like looking for ledger ids that do not exist
and mix existing/non existant ledger ids